Unshaky/ShakyPressPreventer.m

```diff
@@ -57,58 +57,73 @@ - (void)loadIgnoreExternalKeyboard {
     ignoreExternalKeyboard = [defaults boolForKey:@"ignoreExternalKeyboard"]; // default No
 }
 
-- (CGEventRef)filterShakyPressEvent:(CGEventRef)event {
 
-    // keyboard type, dismiss if it is not built-in keyboard
-    if (ignoreExternalKeyboard) {
-        int64_t type = CGEventGetIntegerValueField(event, kCGKeyboardEventKeyboardType);
-        if (type != 58) return event;
-    }
+/** Current version (for fork/pull request): */
+- (CGEventRef)filterShakyPressEvent:(CGEventRef)event {
 
     // The incoming keycode.
     CGKeyCode keyCode = (CGKeyCode)CGEventGetIntegerValueField(event, kCGKeyboardEventKeycode);
     
     // ignore unconfigured keys
     if (keyDelays[keyCode] == 0) return event;
+    
+    // keyboard type, dismiss if it is not built-in keyboard
+    if (ignoreExternalKeyboard) {
```

Comment:
Makes sense. I figured that the minority of people would be using this with an external keyboard, since the butterfly keyboard's issues are the main reason to use it, so I figured that would be the earlier exit.  Maybe not though.